src/test/java/org/mockitousage/bugs/FillInStackTraceScenariosTest.java

```diff
@@ -12,25 +12,40 @@
 import static org.junit.Assert.fail;
 import static org.mockito.Mockito.when;
 
-public class NPEWhenCustomExceptionStackTraceReturnNullTest extends TestBase {
+public class FillInStackTraceScenariosTest extends TestBase {
 
-    @Mock
-    IMethods mock;
+    @Mock IMethods mock;
+
+    private class SomeException extends RuntimeException {}
 
     class NullStackTraceException extends RuntimeException {
-        @Override
         public Exception fillInStackTrace() {
             return null;
         }
     }
 
+    class NewStackTraceException extends RuntimeException {
```

Comment:
Shan't we add a comment that these implementations of `fillInStackTrace` are totally incorrect -- they violate the method contract? Any Throwable implementation must always return a reference to `this` from `fillInStackTrace`.

Reply:
Go ahead! I wanted to preserve current behavior in a test.